cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -312,43 +505,226 @@ def migrate_from_modulestore(
 
     _create_migration_artifacts_incrementally(
         root_migrated_node=root_migrated_node,
-        source=source,
+        source=source_data.source,
         migration=migration,
         status=status,
     )
-
-    block_migrations = ModulestoreBlockMigration.objects.filter(overall_migration=migration)
     status.increment_completed_steps()
 
+    # Forwarding legacy content to migrated content
     status.set_state(MigrationStep.FORWARDING.value)
     if forward_source_to_target:
-        block_sources_to_block_migrations = {
-            block_migration.source: block_migration for block_migration in block_migrations
-        }
-        for block_source, block_migration in block_sources_to_block_migrations.items():
-            block_source.forwarded = block_migration
-            block_source.save()
-
-        source.forwarded = migration
-        source.save()
+        _forwarding_content(migration, source_data)
     status.increment_completed_steps()
 
+    # Populating the collection
     status.set_state(MigrationStep.POPULATING_COLLECTION.value)
     if target_collection:
-        block_target_pks: list[int] = list(
-            ModulestoreBlockMigration.objects.filter(
-                overall_migration=migration
-            ).values_list('target_id', flat=True)
+        _pupulate_collection(user_id, migration)
+    status.increment_completed_steps()
+
+
+@shared_task(base=_BulkMigrationTask, bind=True)
+# Note: The decorator @set_code_owner_attribute cannot be used here because the UserTaskMixin
+#   does stack inspection and can't handle additional decorators.
+def bulk_migrate_from_modulestore(
+    self: _BulkMigrationTask,
+    *,
+    user_id: int,
+    sources_pks: list[int],
+    target_package_pk: int,
+    target_library_key: str,
+    target_collection_pks: list[int | None],
+    create_collections: bool = False,
+    repeat_handling_strategy: str,
+    preserve_url_slugs: bool,
+    composition_level: str,
+    forward_source_to_target: bool,
+) -> None:
+    """
+    Import courses or legacy libraries into a learning package.
```

Comment:
```suggestion
    Import multiple courses or legacy libraries into a learning package.
```
Though for now it's probably more accurate to say "into a content library".

Reply:
Updated https://github.com/openedx/edx-platform/pull/37381/commits/6b653d4005a7612e49dbfe16310d80828eec2fc9